From 364d81afbf4fdc1b3d35c137f02ffa6f1ddd929d Mon Sep 17 00:00:00 2001
From: Debbie Horsfall <debbie.horsfall@arm.com>
Date: Wed, 17 Sep 2025 17:22:21 +0100
Subject: [PATCH] vexpress64: Set the DM_RNG property

Enable the DM_RNG virtio random number generator driver in order to
consume entropy within U-Boot. This allows U-Boot to inject entropy to the
kernel via UEFI, so the kernel can use that early, for instance for
address layout randomisation, or when the kernel does not provide an
entropy driver itself.

Signed-off-by: Debbie Horsfall <debbie.horsfall@arm.com>
Reviewed-by: Andre Przywara <andre.przywara@arm.com>
Upstream-Status: Backport [2767386806c254820ea22a812bb02e5405495a29]
---
 board/armltd/vexpress64/Kconfig | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/board/armltd/vexpress64/Kconfig b/board/armltd/vexpress64/Kconfig
index 7e8709444fe3..610ab0ac37dc 100644
--- a/board/armltd/vexpress64/Kconfig
+++ b/board/armltd/vexpress64/Kconfig
@@ -21,6 +21,8 @@ config VEXPRESS64_BASE_MODEL
 	imply EFI_SET_TIME if DM_RTC
 	select LINUX_KERNEL_IMAGE_HEADER
 	select POSITION_INDEPENDENT
+	imply DM_RNG
+	imply RNG_ARM_RNDR
 
 choice
 	prompt "VExpress64 board variant"
